docs-src/extensionFunctions.adoc

```diff
@@ -0,0 +1,20 @@
+=== Extension functions
+
+Extension functions allow us to attach custom functions to any library or custom code we write. You can extend String, Collection or 
+any other known type with some custom behavior. Given example creates `isContainedBy` extension function that we can directly call on 
+any kind of `T` type, as though it was part of the standard library.
```

Comment:
I've merged it for now but once we get this on the master, if you want a more trivial, let's do that. I'm also trying to remove any text that is strictly unnecessary, i.e. let's keep the focus on annotating lines of code and not convert it into a doc site (if that makes sense :)